src/fidesops/graph/graph_differences.py

```diff
@@ -0,0 +1,200 @@
+from typing import Any, Dict, List, Optional, Set
+
+from fidesops.graph.config import (
+    ROOT_COLLECTION_ADDRESS,
+    TERMINATOR_ADDRESS,
+    CollectionAddress,
+)
+from fidesops.schemas.base_class import BaseSchema
+from fidesops.util.collection_util import Row
+
+GraphRepr = Dict[str, Dict[str, List[str]]]
+
+
+def format_graph_for_caching(
+    env: Dict[CollectionAddress, Any], end_nodes: List[CollectionAddress]
+) -> GraphRepr:
+    """
+    Builds a representation of the current graph (that includes its edges) for caching in Redis.
+
+    Requires the results of traversal.traverse():
+        - the modified `env`
+        - and the outputted end_nodes, which are the final nodes without children
+
+    Maps collections to their upstream dependencies and associated edges. The root is stored as having
+    no upstream collections and the terminator collection has no incoming edges.
+
+    Example:
+    {
+       <collection>: {
+          <upstream collection>: [edge between upstream and current],
+          <another upstream collection>: [edge between another upstream and current]
+       },
+       <root collection>: {},
+       <terminator collection>: {
+            <end collection>: [],
+            <end collection>: []
+       }
+    }
+    """
+    graph_repr: GraphRepr = {
+        collection.value: {
+            upstream_collection_address.value: [str(edge) for edge in edge_list]
+            for upstream_collection_address, edge_list in g_task.incoming_edges_by_collection.items()
+        }
+        for collection, g_task in env.items()
+    }
+    graph_repr[ROOT_COLLECTION_ADDRESS.value] = {}
+    graph_repr[TERMINATOR_ADDRESS.value] = {
+        end_node.value: [] for end_node in end_nodes
+    }
+
+    return graph_repr
+
+
+class GraphDiff(BaseSchema):
+    previous_collections: List[str] = []
+    current_collections: List[str] = []
+    added_collections: List[str] = []
+    removed_collections: List[str] = []
+    added_edges: List[str] = []
+    removed_edges: List[str] = []
+    already_processed_access_collections: List[str] = []
+    already_processed_erasure_collections: List[str] = []
+    skipped_added_edges: List[str] = []
+
+
+class GraphDiffSummary(BaseSchema):
```

Comment:
Hm, well I specifically don't want to send the information from `GraphDiff` to Fideslog, these are details about customer graphs, I just want to send the counts to fideslog to get a high level summary about how their graphs changed, without knowing the internals.

I still had the GraphDiff separated out for testing, so I could have some assurance that the underlying counts were correct.

Reply:
You could still send just the counts to the log but it would take processing to do so simpler in some places, but more complex in others.

Let's go with the way you have it now since we already know it works and don't have to add the processing part.